tests/geo_image_test.py

```diff
@@ -586,3 +586,23 @@ def test_copy_from_extent_geoimage(tmp_path):
                 BytesIO(geoimage.image_data.values).getbuffer().nbytes
                 > BytesIO(geoimage2.image_data.values).getbuffer().nbytes
             )
+
+
+def test_complex_tiff(tmp_path):
+    image_path = tmp_path / r"testtif.tif"
+    workspace_path = tmp_path / r"geo_image_test.geoh5"
+
+    # create a tiff
+    image = Image.fromarray(1000 * np.random.randn(128, 128).astype("float32"))
+    image.save(image_path)
+
+    with Workspace.create(workspace_path) as workspace:
+        # load image
+        geoimage = GeoImage.create(workspace, name="test_area", image=str(image_path))
+
+        grid = workspace.get_entity("test_area_grid2d")[0]
+
+        assert all(
+            np.array(geoimage.image)[::-1].flatten()
+            == grid.get_data("band[0]")[0].values
+        )
```

Comment:
Can we add a check for multi-bands too?

Reply:
PIL images can't handle >4 bands. 
That's why people are using Gdal

Reply:
Actually, I can play with more than 4 bands, but that's another level of complexity. 
I created an issue to refactor geoimage for tiff.

Reply:
I'll test for 4 bands here, and we will implement n bands later

Reply:
Ok doke